From 5c76f3356fe2178677019e0415151e64605411c4 Mon Sep 17 00:00:00 2001
From: Karl Dietz <dekarl@mythtv.org>
Date: Wed, 23 Jul 2014 11:15:24 +0200
Subject: [PATCH] Attempt to fix segfault in EITCache assumed to be due to
 QMap race

Fixes #12219
(cherry picked from commit 477c3f2b1cd3bcb38c5cfd9a3a934a95e1a13beb)
---
 mythtv/libs/libmythtv/eitcache.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/eitcache.cpp b/mythtv/libs/libmythtv/eitcache.cpp
index c087d9c..ae7c98d 100644
--- a/mythtv/libs/libmythtv/eitcache.cpp
+++ b/mythtv/libs/libmythtv/eitcache.cpp
@@ -274,14 +274,14 @@ void EITCache::WriteChannelToDB(QStringList &value_clauses, uint chanid)
                 updated++;
                 *it &= ~(uint64_t)0 >> 1; // mark as synced
             }
+            ++it;
         }
         else
         {
             // Event is too old; remove from eit cache in memory
-            eventMap->remove(it.key());
+            it = eventMap->erase(it);
             removed++;
         }
-        ++it;
     }
     unlock_channel(chanid, updated);
 
-- 
1.7.10.2

